tools/scripts/generateSoundMetadata.rb

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env ruby
+#
+# Given a list of mp3 sounds, generates a matching set of json files that
+# consist of sound metadata for use in the Game Lab/App Lab sound library,
+# ready to upload to S3.
+#
+# Script adapted from generateSingleFrameAnimationMetadata.rb
+#
+# For usage, run ./generateSoundMetadata.rb --help
+#
+# See also: The Sound Library Tech Spec (requires login):
+# https://docs.google.com/document/d/1Oj--H-xwrK3u4A0L5ML73n8XDutsH5n2i8caGRc3NL8/edit
+#
+require 'optparse'
+require 'parallel'
+require 'taglib'
+require_relative '../../deployment'
+require_relative '../../lib/cdo/cdo_cli'
+include CdoCli
+
+DEFAULT_S3_BUCKET = 'cdo-sound-library'.freeze
+DEFAULT_OUTPUT_FILE = 'apps/src/code-studio/soundLibrary.json'.freeze
+
+class MetadataBuilder
+  def initialize(options)
+    @options = options
+  end
+
+  def build_sound_metadatas(files)
+    unique_files = files.uniq
+
+    # Load each file and generate a metadata file for it
+    Parallel.map(unique_files) do |filename|
+      dirname = File.dirname(filename)
+      basename = File.basename(filename, '.mp3')
+      TagLib::MPEG::File.open(filename) do |mp3_file|
+        metadata = {}
+        metadata['name'] = basename
+        metadata['aliases'] = @options[:aliases] || []
+        metadata['time'] = mp3_file.audio_properties.length
```

Comment:
TagLib allows us to get the duration of the mp3 files in seconds. Many of the Kenney.nl sounds are less than 1 second and show up as zero but I had trouble finding a ruby library that parsed sound in miliseconds, so I'll go with this for now.